From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Taiyou06 <kaandindar21@gmail.com>
Date: Thu, 7 Nov 2024 19:45:31 +0100
Subject: [PATCH] C2ME: Reduce Allocations

This patch is based on the following mixin:
"com/ishland/c2me/opts/allocs/mixin/object_pooling_caching/MixinOreFeature.java"
By: ishland <ishlandmc@yeah.net>
As part of: C2ME (https://github.com/RelativityMC/C2ME-fabric)
Licensed under: MIT (https://opensource.org/licenses/MIT)

// SYNC SOURCE START
rev: 8138acfb32daf5d1b264780b6436eb8ff958c1b9
branch: dev/26.2.0
repo: https://github.com/RelativityMC/C2ME-fabric.git
track:
- "c2me-opts-allocs/src/main/java/com/ishland/c2me/opts/allocs/mixin/object_pooling_caching/MixinOreFeature.java"
- "c2me-opts-allocs/src/main/java/com/ishland/c2me/opts/allocs/common/ObjectCachingUtils.java"
// SYNC SOURCE END

diff --git a/net/minecraft/world/level/levelgen/feature/OreFeature.java b/net/minecraft/world/level/levelgen/feature/OreFeature.java
index e6aee6a38f45b47413aa3bbddf3632bc0c6a93f5..7e902e3d4d25f808e4907d051942cb8d2aa1e515 100644
--- a/net/minecraft/world/level/levelgen/feature/OreFeature.java
+++ b/net/minecraft/world/level/levelgen/feature/OreFeature.java
@@ -80,7 +80,7 @@ public class OreFeature extends AbstractOreFeature {
         final int sizeY
     ) {
         int placed = 0;
-        BitSet tested = new BitSet(sizeXZ * sizeY * sizeXZ);
+        BitSet tested = org.dreeam.leaf.util.cache.CachedOrNewBitsGetter.getCachedOrNewBitSet(sizeXZ * sizeY * sizeXZ); // Leaf - C2ME - Reduce Allocations
         BlockPos.MutableBlockPos orePos = new BlockPos.MutableBlockPos();
         double[] data = new double[this.size * 4];
 
